internal/resources/webform/webform.js

```diff
@@ -2435,48 +2449,67 @@ window.initGRPCForm = function(services, invokeURI, metadataURI, debug, headers)
     const clearHistory = () => {
         if (confirm('Are you sure you wish to delete all history? This action is permanent and cannot be undone.')) {
             history = [];
+            preloadedHistory = [];
```

Comment:
I don't think this is necessary. Instead, when you load the server history, if it is not empty, just merge it into `history`. I think that would let you undo some of the changes/refactoring in `updateHistoryUI()` below and also return to the much simpler handling of `history` with a single index. That would allow this PR to become smaller and much simpler.

Reply:
Merging was my initial approach, but it ended up in a positive feedback loop that increases history with every page reload.

This is a pretty rough draft I used to get an idea of what changes would be needed. My intent is to mirror our process for HTTP: each service has a postman collection saved in the source code so new devs or clients of our service can quickly get up to speed with the api.

It would be beneficial to separate history from "official" examples, the expectation is that examples are known to work (or fail in a documented fashion) while history is simply there to give access to old calls. In the current draft the separation between examples and history is highlighted through the gray background in the image below.

![Screenshot 2020-10-23 at 09 56 26](https://user-images.githubusercontent.com/35903525/96972151-67cec500-1516-11eb-996a-a1c9a9433e6a.png)

A more elaborate solution would be to add explicit examples support, in the UI this would be a list on the left side of the request tab. Clicking on an example the request form is populated with the data from the example, history will work unchanged as before. From the back end code, we'll add a dedicated `/examples` endpoint that can be populated by an option, e.g. `standalone.WithExamples(exablesBlob)`, if the option is not provided `/examples` will return an empty array which will lead the UI to not create the examples list from the examples tab.

Data schema will use the same format as history plus a description part where we can add a short name and detailed description.

Reply:
Ok. I didn't think through it enough to foresee the ever-enlarging-history problem :)

Reply:
For now, how about renaming `preloadedHistory` -> `examples`? And I think it may also make sense to move them to a separate HTML table with their own heading, to make it more clear what they are (instead of relying on just a gray background).

Reply:
I've updated the UI to handle examples in a cleaner way. If `standalone.WithExamples` is used the UI will look like the image. If no examples are present the examples column is hidden. This approach is similar to postman and other similar tools so devs should understand it easily.
<img width="1208" alt="Screenshot 2020-10-23 at 17 44 47" src="https://user-images.githubusercontent.com/35903525/97024758-74274200-1557-11eb-8f3a-e19df350bff8.png">

Reply:
I also plan to add some documentation once the flow is in good state.